src/synapse/api.py

```diff
@@ -69,36 +77,39 @@ def register_user(registration_shared_secret: str, user: User) -> None:
 
     Raises:
         NetworkError: if there was an error registering the user.
+        GetNonceError: if there was an error while getting nonce.
     """
-    # get nonce
-    nonce = _get_nonce()
-
-    # generate mac
-    hex_mac = _generate_mac(
-        shared_secret=registration_shared_secret,
-        nonce=nonce,
-        user=user.username,
-        password=user.password,
-        admin=user.admin,
-    )
-    data = {
-        "nonce": nonce,
-        "username": user.username,
-        "password": user.password,
-        "mac": hex_mac,
-        "admin": user.admin,
-    }
-    # finally register user
     try:
+        # get nonce
+        nonce = _get_nonce()
+
+        # generate mac
+        hex_mac = _generate_mac(
+            shared_secret=registration_shared_secret,
+            nonce=nonce,
+            user=user.username,
+            password=user.password,
+            admin=user.admin,
+        )
+        data = {
+            "nonce": nonce,
+            "username": user.username,
+            "password": user.password,
+            "mac": hex_mac,
+            "admin": user.admin,
+        }
+        # finally register user
         res = requests.post(REGISTER_URL, json=data, timeout=5)
         res.raise_for_status()
-    except (
-        requests.exceptions.ConnectionError,
-        requests.exceptions.Timeout,
-        requests.exceptions.HTTPError,
-    ) as exc:
-        logger.exception("Failed to request %s : %r", REGISTER_URL, exc)
-        raise NetworkError(f"Failed to request {REGISTER_URL}.") from exc
+    except (requests.exceptions.ConnectionError, requests.exceptions.Timeout) as exc:
+        logger.exception("Failed to connect to %s: %r", REGISTER_URL, exc)
+        raise NetworkError(f"Failed to connect to {REGISTER_URL}.") from exc
+    except requests.exceptions.HTTPError as exc:
+        logger.exception("HTTP error from %s: %r", REGISTER_URL, exc)
+        raise NetworkError(f"HTTP error from {REGISTER_URL}.") from exc
```

Comment:
Should we scope catching these exceptions to specifically the lines where the can occur? I.e., lines 102 and 103?